pkg/kn/commands/channel/flags.go

```diff
@@ -120,3 +125,64 @@ func printChannelListWithNamespace(channelList *messagingv1beta1.ChannelList, op
 
 	return append(rows, others...), nil
 }
+
+// ListTypesHandlers handles printing human readable table for `kn channel list-types`
+func ListTypesHandlers(h printers.PrintHandler) {
+	channelTypesColumnDefinitions := []metav1beta1.TableColumnDefinition{
+		{Name: "Type", Type: "string", Description: "Kind / Type of the channel type", Priority: 1},
+		{Name: "Group", Type: "string", Description: "Group of the channel type", Priority: 1},
+		{Name: "Version", Type: "string", Description: "Version of the channel type", Priority: 1},
+	}
+	h.TableHandler(channelTypesColumnDefinitions, printChannelTypes)
+	h.TableHandler(channelTypesColumnDefinitions, printChannelTypesList)
+}
+
+// printChannelTypes populates a single row of channel types list table
+func printChannelTypes(channelType unstructured.Unstructured, options printers.PrintOptions) ([]metav1beta1.TableRow, error) {
+	name := channelType.GetName()
+	content := channelType.UnstructuredContent()
+	version := channelType.GroupVersionKind().Version
+	kind, found, err := unstructured.NestedString(content, "spec", "names", "kind")
+	if err != nil {
+		return nil, err
+	}
+	if !found {
+		return nil, fmt.Errorf("can't find specs.names.kind for %s", name)
+	}
+
+	group, found, err := unstructured.NestedString(content, "spec", "group")
+	if err != nil {
+		return nil, err
+	}
+
+	if !found {
+		return nil, fmt.Errorf("can't find specs.group for %s", name)
+	}
+
+	row := metav1beta1.TableRow{
+		Object: runtime.RawExtension{Object: &channelType},
+	}
+	row.Cells = append(row.Cells, kind, group, version)
+
+	return []metav1beta1.TableRow{row}, nil
+}
+
+// printChannelTypesList populates the channel types list table rows
+func printChannelTypesList(channelTypesList *unstructured.UnstructuredList, options printers.PrintOptions) ([]metav1beta1.TableRow, error) {
+	rows := make([]metav1beta1.TableRow, 0, len(channelTypesList.Items))
+
+	sort.SliceStable(channelTypesList.Items, func(i, j int) bool {
+		return channelTypesList.Items[i].GetName() < channelTypesList.Items[j].GetName()
+	})
+	for _, item := range channelTypesList.Items {
+		if item.GetName() != channelName {
```

Comment:
Here you exclude `channels.messaging.knative.dev` manually. But it will bring inconsistency when you call `kn channel list-types -o name` or `kn channel list-types -o yaml`. Here is the test result in my environment:

```
$ ./kn channel list-types -o name
customresourcedefinition.apiextensions.k8s.io/channels.messaging.knative.dev
customresourcedefinition.apiextensions.k8s.io/inmemorychannels.messaging.knative.dev
customresourcedefinition.apiextensions.k8s.io/kafkachannels.messaging.knative.dev
$ ./kn channel list-types
TYPE              GROUP                   VERSION
InMemoryChannel   messaging.knative.dev   v1beta1
KafkaChannel      messaging.knative.dev   v1beta1
```

Please address the issue. If you are not able to remove `customresourcedefinition.apiextensions.k8s.io/channels.messaging.knative.dev` from the output of `kn channel list-types -o name`, maybe you have to add this item back.